File: 220/js/observer/stockchecker_spec.js
require('./stockchecker')

describe("StockChecker", function() {
  it("informs observer of changes in stock level", function() {
    var stockchecker = new StockChecker(5);
    var observer = new StockObserver();
    stockchecker.addObserver(observer);
    stockchecker.buyItem();
    expect(observer.latestNotification()).toEqual(4);
  });
  
  it("can have multiple observers", function() {
   	 var stockchecker = new StockChecker(5);
	 var observer1 = new StockObserver();
     var observer2 = new StockObserver();
	 stockchecker.addObserver(observer1);
	 stockchecker.addObserver(observer2);
     stockchecker.buyItem();
     expect(observer1.latestNotification()).toEqual(4);
     expect(observer2.latestNotification()).toEqual(4);
  });
});

File: 220/ruby/observer/stockchecker_spec.rb
require './stockchecker'

describe StockChecker do
  it "informs observer of changes in stock level" do
    stockchecker = StockChecker.new(5)
    observer = StockObserver.new
    stockchecker.add_observer(observer)
    stockchecker.buy_item
    observer.latest_notification.should == 4
  end
  
  it "can have multiple observers" do
    stockchecker = StockChecker.new(5)
    observer1 = StockObserver.new
    observer2 = StockObserver.new
    stockchecker.add_observer(observer1)  
    stockchecker.add_observer(observer2)
    stockchecker.buy_item
    observer1.latest_notification.should == 4
    observer2.latest_notification.should == 4
  end
  
end
